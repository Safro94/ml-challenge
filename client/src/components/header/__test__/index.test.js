import React from 'react';

import { render } from '@testing-library/react';
import { screen } from '@testing-library/dom';
import userEvent from '@testing-library/user-event';
import { Link } from 'react-router-dom';

import Header from '../';

jest.mock('react-router-dom', () => ({
	useHistory: () => ({ push: {} }),
	Link: jest.fn(({ children, ...rest }) => <a {...rest}>{children}</a>),
}));

const mockSetSearchTerm = jest.fn();
const mockSetResult = jest.fn();
jest.mock('hooks/application', () => ({
	useApplication: jest.fn(() => ({
		setSearchTerm: mockSetSearchTerm,
		setResult: mockSetResult,
	})),
}));

describe('Header', () => {
	it('should render OK', () => {
		//Act
		const { container } = render(<Header />);

		//Assert
		expect(container).toMatchSnapshot();
	});

	it('should go to the results page and when the logo is clicked', async () => {
		//Act
		render(<Header />);
		userEvent.click(screen.getByTestId('logo'));

		//Assert
		expect(Link).toHaveBeenCalledTimes(1);
		expect(Link.mock.calls[0][0].to).toBe('/');

		expect(mockSetResult).toHaveBeenCalledTimes(1);
		expect(mockSetResult).toHaveBeenCalledWith({ items: [], categories: [] });

		expect(mockSetSearchTerm).toHaveBeenCalledTimes(1);
		expect(mockSetSearchTerm).toHaveBeenCalledWith('');
	});
});
