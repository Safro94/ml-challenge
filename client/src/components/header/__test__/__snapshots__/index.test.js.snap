// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render OK 1`] = `
<div>
  <header>
    <div
      class="root"
    >
      <div
        class="container"
      >
        <div>
          <a
            data-testid="logo"
            to="/"
          >
            <img
              alt="Logo"
              class="logo"
              src="/assets/Logo_ML@2x.png.png.png"
            />
          </a>
        </div>
        <div>
          <form
            class="form"
          >
            <input
              class="input"
              placeholder="Nunca dejes de buscar"
              title="Escribi tu texto de busqueda aqui"
              type="text"
              value=""
            />
            <button
              class="button button"
              disabled=""
              type="submit"
            >
              <img
                alt="Buscar"
                class="icon"
                src="/assets/ic_Search@2x.png.png.png"
              />
            </button>
          </form>
        </div>
      </div>
    </div>
  </header>
</div>
`;
